### PR TITLE
Optimise reloads

### PR DESCRIPTION
**Pull request by nordfjord**
- perf(message-db): optimise reloads
- perf(libsql): optimise reloads

### CODE REVIEW

File: packages/libsql/src/lib/Category.ts [hunk 1]
```diff
@@ -252,18 +253,25 @@ class InternalCategory<Event, State, Context>
   }
 
   async reload(streamId: Equinox.StreamId, _requireLeader: boolean, t: TokenAndState<State>) {
-    if (this.access.type === "RollingState") {
-      return this.load(streamId, 0)
+    switch (this.access.type) {
+      // These strategies all have it in common that their state 
+      // can be reconstituted in a single round trip to the database
+      // as such is it as cheap to load them as it is to reload them
```
Comment (nordfjord) [inline packages/libsql/src/lib/Category.ts:259]: possible caveat on snapshot, where the reload could be sending more data over the wire in cases where the cached value is the same as what's in the cache

Comment (bartelink) [inline packages/libsql/src/lib/Category.ts:259]:
```suggestion
      // as such, is it as cheap to load them as it is to reload them
```
But maybe add even more explicit stuff?
// for this reason, unlike with some other stores, there's no particular win to be had in terms of
// using a more effficient index based on some cached token or etag, reducing roundtrips etc.

File: packages/libsql/src/lib/Category.ts [hunk 2]
```diff
@@ -252,18 +253,25 @@ class InternalCategory<Event, State, Context>
...
+      case "LatestKnownEvent":
+        return this.load(streamId, 0)
+      case "Unoptimized":
+        const streamName = Equinox.StreamName.create(this.categoryName, streamId)
+        const [token, state] = await this.context.loadBatched(
+          streamName,
+          this.codec.decode,
+          this.fold,
+          t.state,
+          Token.version(t.token),
```
Comment (bartelink) [inline packages/libsql/src/lib/Category.ts:271]: maybe a helper for getting `[token, state]` from `t` would make things clearer (if there's only one or two never mind; this is obviously not exactly cryptic)

File: packages/message-db/message-db/src/lib/Category.ts [hunk 3]
```diff
@@ -194,6 +194,8 @@ class InternalCategory<Event, State, Context>
   private async loadAlgorithm(
     streamId: Equinox.StreamId,
     requireLeader: boolean,
+    initial: State,
+    version = 0n,
```
Comment (bartelink) [inline packages/message-db/message-db/src/lib/Category.ts:198]: not sure about defalt arg as you didnt do it for loadbatched in the other impl

Comment (nordfjord) [reply packages/message-db/message-db/src/lib/Category.ts:189]: fair, went through and reduced the line count through "uglification" as well

File: packages/message-db/message-db/src/lib/Category.ts [hunk 4]
```diff
@@ -206,87 +197,43 @@ class InternalCategory<Event, State, Context>
...
-        ;[token, state] = await this.context.loadBatched(
-          streamName,
-          requireLeader,
-          this.codec.decode,
-          this.fold,
-          t.state,
-          Token.version(t.token),
-        )
+        // prettier-ignore
+        result = await this.context.loadBatched(streamName, requireLeader, this.codec.decode, this.fold, t.state, Token.version(t.token))
```
Comment (bartelink) [inline packages/message-db/message-db/src/lib/Category.ts:233]: could the last 2 args become a TokenAndState?

**Pull request merged**